src/main/target/SPEEDYBEEF7/target.h

```diff
@@ -49,6 +49,8 @@
 #define USE_GYRO_SPI_ICM20689
 #define USE_GYRO_SPI_MPU6000
 
+#define USE_RANGEFINDER
```

Comment:
Already defined in unified target for F7 but not for other targets.

Reply:
it is defined with limits but enabled with those limitation in speedybeef7v2 so i had to add in stm32f7x2 which is base target for speedybeef7V2

Reply:
Added `#define USE_RANGEFINDER` to line 411 in `target/common_pre.h` and `make all` compiled.

Reply:
super my friend you are awesome

Reply:
> Added `#define USE_RANGEFINDER` to line 411 in `target/common_pre.h` and `make all` compiled.

what is the difference between add to 'target/common_pre.h'  and  '' src/main/target/STM32F7X2/target.h" if I use a F722 flight controller

Reply:
Simply add `USE_RANGEFINDER` as custom define `RANGEFINDER` while flashing as it is not part of any board configuration and should not be added to platform or MCU definitions either.

Reply:
> Simply add `USE_RANGEFINDER` as custom define `RANGEFINDER` while flashing as it is not part of any board configuration and should not be added to platform or MCU definitions either.

Thx for real quick reply. I am not sure if I get you. I wanted to add tfmini rangefinder to  betaflight runs on F722. Now I added #define USE_RANGEFINDER and #define USE_RANGEFINDER_TF in src/main/target/STM32F7X2/target.h (Frimware 4.4) but it doesn't work. I'm not asking any pull request or merge stuff. So, if it can work, it is fine to me.

Reply:
If not using configurator but want to build locally for firmware 4.4.3 you could use something like

```
make STM32F7X2 EXTRA_FLAGS="-D'BUILD_KEY=93ca8096f23fa70e42d10717c2014867' -D'RELEASE_NAME=4.4.3' -D'BOARD_NAME=SPEEDYBEEF7V2' -D'MANUFACTURER_ID=SPBE' -DCLOUD_BUILD -DUSE_ACC -DUSE_ACC_SPI_MPU6000 -DUSE_BARO -DUSE_BARO_BMP280 -DUSE_DSHOT -DUSE_FLASH -DUSE_FLASH_W25Q128FV -DUSE_GPS -DUSE_GPS_PLUS_CODES -DUSE_GYRO -DUSE_GYRO_SPI_MPU6000 -DUSE_LED_STRIP -DUSE_MAX7456 -DUSE_OSD -DUSE_OSD_SD -DUSE_PINIO -DUSE_RANGEFINDER -DUSE_SERIALRX -DUSE_SERIALRX_CRSF -DUSE_TELEMETRY -DUSE_TELEMETRY_CRSF -DUSE_VTX" 
```

Reply:
I followed your answer and added #define USE_RANGEFINDER and #define USE_RANGEFINDER_TF in src/main/target/STM32F7X2/target.h, added $(addprefix drivers/rangefinder/,$(notdir $(wildcard $(SRC_DIR)/drivers/rangefinder/*.c))) in src\main\target\STM32F7X2\target.mk, and then made it. Then I entered the configurator to change the .hex file, set the port, and finally typed ‘feature RANGEFINDER’, ‘set rangefinder_hardware = TFMINI’, ‘serial 1 32768 115200 57600 0 115200`, ‘save’ in the CLI. But it doesn’t seem to work, the sonar logo is still gray.
<img width="1280" alt="屏幕截图 2024-08-08 131836" src="https://github.com/user-attachments/assets/56db5d6b-ada0-495c-98e3-c38a7864a824">
<img width="1259" alt="屏幕截图 2024-08-08 132122" src="https://github.com/user-attachments/assets/a0927e36-134b-4180-9d8a-d90bc5e09a27">
<img width="1280" alt="屏幕截图 2024-08-08 132252" src="https://github.com/user-attachments/assets/00b80504-6a1c-4e10-8a9f-f0d1cf289cf2">